src/domain/authentication/refreshAuth.ts

```diff
@@ -0,0 +1,10 @@
+import authService from './authService';
+import isUnexpiredToken from './isUnexpiredToken';
+
+export const refreshAuth = async (): Promise<void> => {
+  const accessToken = authService.getToken();
+  const refreshToken = authService.getRefreshToken();
+  if (!isUnexpiredToken(accessToken) && isUnexpiredToken(refreshToken)) {
+    await authService.renewToken();
+  }
+};
```

Comment:
the silent renew might have always done the refresh already at this point when the accessToken is expired, but the refreshToken is not. At least it should have, because otherwise the authorization would have already ended.

However, it seems like this is how the Marmelab suggests you to do in https://marmelab.com/react-admin/addRefreshAuthToAuthProvider.html, so I guess it's totally fine with the React-Admin.